forch/faucet_state_collector.py

```diff
@@ -780,14 +780,15 @@ def _get_egress_state_detail(self, links):
         links_status = set()
         egress_name = None
         for key, status in links.items():
-            links_status.add(LacpState.LacpState.Value(status.get(LINK_STATE)))
-            if status.get(LINK_STATE) == LacpState.LacpState.Name(LacpState.active):
+            link_state = status.get(LINK_STATE)
+            links_status.add(LacpState.LacpState.Value(link_state))
+            if link_state == LacpState.LacpState.Name(LacpState.active):
                 egress_name = key
-            if status.get(LINK_STATE) == LacpState.LacpState.Name(LacpState.down):
+            elif link_state == LacpState.LacpState.Name(LacpState.down):
                 link_down = key
-        if links_status == set([LacpState.active, LacpState.up]):
+        if links_status == set([LacpState.active, LacpState.up, LacpState.unknown]):
```

Comment:
Current implementation assumes that this event gets called only for egress links but that is not true anymore.